packages/fe-fpm-writer/test/integration/custom-page-app.test.ts

```diff
@@ -0,0 +1,109 @@
+        const tsConfig = {
+            path: join(testOutput, 'ts')
+        };
+        const configs: { path: string }[] = [basicConfig, tsConfig];
+
+        beforeAll(() => {
+            fs.copy(join(testInput, 'js'), basicConfig.path, { globOptions: { dot: true } });
+            fs.copy(join(testInput, 'ts'), tsConfig.path, { globOptions: { dot: true } });
+        });
+
+        test.each(configs)('generateBuildingBlock:FilterBar in custom page', (config) => {
+            generateBuildingBlock<FilterBar>(
+                config.path,
+                {
+                    viewOrFragmentPath: join('webapp/ext/main/Main.view.xml'),
+                    aggregationPath: `/mvc:View/*[local-name()='Page']/*[local-name()='content']`,
+                    buildingBlockData: {
+                        id: 'testFilterBar',
+                        buildingBlockType: BuildingBlockType.FilterBar,
+                        metaPath: '@com.sap.vocabularies.UI.v1.SelectionFields'
+                    }
+                },
+                fs
+            );
+        });
+
+        test.each(configs)('generateBuildingBlock:Chart in custom page', (config) => {
+            generateBuildingBlock<Chart>(
+                config.path,
+                {
+                    viewOrFragmentPath: join('webapp/ext/main/Main.view.xml'),
+                    aggregationPath: `/mvc:View/*[local-name()='Page']/*[local-name()='content']`,
+                    buildingBlockData: {
+                        id: 'testChart',
+                        buildingBlockType: BuildingBlockType.Chart,
+                        metaPath: '@com.sap.vocabularies.UI.v1.Chart#testChart',
+                        filterBar: 'testFilterBar',
+                        personalization: 'Type,Item,Sort',
+                        selectionMode: 'Single'
+                    }
+                },
+                fs
+            );
+        });
+
+        test.each(configs)('generateBuildingBlock:Table in custom page', (config) => {
+            generateBuildingBlock<Table>(
+                config.path,
+                {
+                    viewOrFragmentPath: join('webapp/ext/main/Main.view.xml'),
+                    aggregationPath: `/mvc:View/*[local-name()='Page']/*[local-name()='content']`,
+                    buildingBlockData: {
+                        id: 'testTable',
+                        buildingBlockType: BuildingBlockType.Table,
+                        metaPath: '@com.sap.vocabularies.UI.v1.LineItem',
+                        busy: false,
+                        enableAutoColumnWidth: true,
+                        enableExport: true,
+                        enableFullScreen: true,
+                        enablePaste: true,
+                        filterBar: 'testFilterBar',
+                        header: 'Test Header',
+                        headerVisible: true,
+                        isSearchable: true,
+                        menu: 'testMenuActionKey',
```

Comment:
I have to check in FE but i don't think menu should be in there (or even what it is), can you please remove it ?

Reply:
yes it is useless, the doc should be adjusted on our end

Reply:
@nlunets: I removed the menu property.